astropy/io/ascii/ecsv.py

```diff
@@ -25,6 +25,31 @@
     'float128', 'string')
 
 
+def _old_version_workarounds(version):
```

Comment:
I'd rather not invent this general infrastructure before we actually need it. In this case I think we can eliminate this whole function and replace it with `and self.ecsv_version > (0, 9, 0)` and a long comment that explains *why* we are doing this. See below.

Reply:
Agreed I'll remove this. I was worried about what seems like a hack which will we removed in the future getting lost, but I agree that a comment serves this better.